sdk/storage/azure_storage_blob/tests/blob_client.rs

```diff
@@ -1,50 +1,90 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-use azure_core_test::recorded;
-use azure_identity::DefaultAzureCredential;
-use azure_storage_blob::{BlobBlobClientGetPropertiesOptions, BlobClient};
+use azure_core::{headers::HeaderName, Bytes, RequestContent, StatusCode};
+use azure_core_test::{recorded, TestContext};
+use azure_storage_blob::{
+    clients::{BlobClient, ContainerClient},
+    models::{BlobBlobClientDownloadOptions, BlobBlobClientGetPropertiesOptions, BlobType},
+    BlobClientOptions,
+};
 use std::{env, error::Error};
 
-#[recorded::test(live)]
-async fn test_get_blob_properties() -> Result<(), Box<dyn Error>> {
+#[recorded::test]
+async fn test_get_blob_properties(ctx: TestContext) -> Result<(), Box<dyn Error>> {
+    // Recording Setup
+    let recording = ctx.recording();
+    let mut options = BlobClientOptions::default();
+    recording.instrument(&mut options.client_options);
+
     // Setup
-    let storage_account_name = env::var("AZURE_STORAGE_ACCOUNT_NAME")
-        .expect("Failed to get environment variable: AZURE_STORAGE_ACCOUNT_NAME");
-    let endpoint = format!("https://{}.blob.core.windows.net/", storage_account_name);
-    let credential = DefaultAzureCredential::new()?;
+    let endpoint = format!(
+        "https://{}.blob.core.windows.net/",
+        recording.var("AZURE_STORAGE_ACCOUNT_NAME", None).as_str()
+    );
 
     // Act
+    let container_client = ContainerClient::new(
+        &endpoint,
+        String::from("testcontainer1"),
+        recording.credential(),
+        Some(options.clone()),
+    )?;
+    container_client.create_container(None).await?;
+
     let blob_client = BlobClient::new(
         &endpoint,
-        String::from("testcontainer"),
+        String::from("testcontainer1"),
```

Comment:
Hmm. I should probably add support to generate random names. In the meantime, though, I recommend you use names based on  your test name for easy identification.

Reply:
I came up with this in a separate branch when testing out test proxy:
```rust
pub fn get_name(recording: &Recording, resource_type: &str) -> String {
    let rand: u128 = recording.random();
    format!("{}{}{}", "test", resource_type, Uuid::from_u128(rand))
}
```

But would be happy to just consume some common utility instead of writing a helper specific to Storage (since I could see this being beneficial to others as well)

Reply:
#2265 is merged so you can use that now to generate seeded random names like in other languages.

Reply:
Awesome, thanks for the quick turnaround on that Heath! I will completely switch to this utility in the following PR when we revisit our tests + add new tests for the other APIs.